test/framework/labels.go

```diff
@@ -18,11 +19,31 @@ func ValidateControlPlaneLabels(controlPlane v1alpha1.ControlPlaneConfiguration,
 	return validateLabels(controlPlane.Labels, node)
 }
 
+// ValidateControlPlaneFailureDomainLabels validate if Cloudstack provider replaces ds.meta_data.failuredomain with proper failuredomain name
+// in control plane node label 'cluster.x-k8s.io/failure-domain'
+func ValidateControlPlaneFailureDomainLabels(controlPlane v1alpha1.ControlPlaneConfiguration, node corev1.Node) error {
+	if controlPlane.MachineGroupRef.Kind == "CloudStackMachineConfig" {
```

Comment:
why do we need these checks for MachineConfig kind?

Reply:
ds.meta_data.failuredomain is keyword only for cloudstack provider. For other providers like vsphere, it will not be replaced with failuredomain.

Reply:
That makes sense, but it seems like other providers wouldn't call this function in the first place, so the extra check is unnecessary.

Reply:
In that case, we should handle the precondition failure case, and return some form of error saying it's only for CloudStack machines

Reply:
add error handling for non cloudstackmachineConfig.